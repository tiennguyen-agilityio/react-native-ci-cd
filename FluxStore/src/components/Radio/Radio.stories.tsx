import {useState} from 'react';
import {Meta, StoryObj} from '@storybook/react';

import Radio from '.';

const meta: Meta<typeof Radio> = {
  title: 'Components/Radio',
  component: Radio,
};

export default meta;

type Story = StoryObj<typeof Radio>;

const Render = (props: {size?: number; borderWidth?: number}) => {
  const [selected, setSelected] = useState(false);
  const handleChange = () => setSelected(prev => !prev);
  return <Radio selected={selected} onPress={handleChange} {...props} />;
};

export const Default: Story = {
  render: Render,
};

export const Large: Story = {
  render: () => <Render size={30} borderWidth={10} />,
};
