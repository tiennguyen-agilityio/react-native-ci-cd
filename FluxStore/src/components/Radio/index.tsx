import {memo, useMemo} from 'react';
import {View, TouchableOpacity, ViewStyle} from 'react-native';
import {useThemeStore} from '@/stores';

interface RadioProps {
  selected?: boolean;
  size?: number;
  borderWidth?: number;
  onPress?: () => void;
}

const Radio = ({selected = false, onPress, size = 23, borderWidth = 7}: RadioProps) => {
  const {theme} = useThemeStore();

  const radioStyles: ViewStyle = useMemo(
    () => ({
      width: size,
      height: size,
      borderRadius: size / 2,
      borderColor: selected ? theme.background.radio : theme.border.quaternary,
      borderWidth: selected ? borderWidth : 1,
    }),
    [selected, size, theme.background.radio, theme.border, borderWidth],
  );

  return (
    <TouchableOpacity activeOpacity={0.8} onPress={onPress}>
      <View style={radioStyles} />
    </TouchableOpacity>
  );
};

export default memo(Radio);
