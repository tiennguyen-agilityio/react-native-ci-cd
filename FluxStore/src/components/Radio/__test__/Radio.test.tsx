import {render} from '@testing-library/react-native';

import Radio from '..';

const onPress = jest.fn();

describe('Button', () => {
  const props = {
    selected: false,
    onPress,
  };

  it('should render Radio component with default props', () => {
    const {toJSON} = render(<Radio {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render Radio component with size', () => {
    const {toJSON} = render(<Radio size={30} borderWidth={10} {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with selected true', () => {
    const {toJSON} = render(<Radio {...props} selected />);
    expect(toJSON()).toMatchSnapshot();
  });
});
