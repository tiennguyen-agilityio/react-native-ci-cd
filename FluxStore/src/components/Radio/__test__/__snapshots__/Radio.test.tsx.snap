// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with selected true 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "borderColor": "#508a7b",
        "borderRadius": 11.5,
        "borderWidth": 7,
        "height": 23,
        "width": 23,
      }
    }
  />
</View>
`;

exports[`Button should render Radio component with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "borderColor": "#bebfc4",
        "borderRadius": 11.5,
        "borderWidth": 1,
        "height": 23,
        "width": 23,
      }
    }
  />
</View>
`;

exports[`Button should render Radio component with size 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "borderColor": "#bebfc4",
        "borderRadius": 15,
        "borderWidth": 1,
        "height": 30,
        "width": 30,
      }
    }
  />
</View>
`;
